### PR TITLE
Improve type annotations for `distributed.core.Server`

### PR DESCRIPTION
**Pull request by hendrikmakait**
Closes #xxxx

- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

File: distributed/scheduler.py [hunk 1]
```diff
@@ -5863,7 +5863,7 @@ def worker_send(self, worker: str, msg: dict[str, Any]) -> None:
             stream_comms[worker].send(msg)
         except (CommClosedError, AttributeError):
             self._ongoing_background_tasks.call_soon(
-                self.remove_worker,
+                self.remove_worker,  # type: ignore[arg-type]
```
Comment (hendrikmakait) [inline distributed/scheduler.py:5866]: Type annotations on the `AsyncTaskGroup` are too narrow.

File: distributed/scheduler.py [hunk 2]
```diff
@@ -5909,7 +5909,7 @@ def send_all(self, client_msgs: Msgs, worker_msgs: Msgs) -> None:
                 pass
             except (CommClosedError, AttributeError):
                 self._ongoing_background_tasks.call_soon(
-                    self.remove_worker,
+                    self.remove_worker,  # type: ignore[arg-type]
```
Comment (hendrikmakait) [inline distributed/scheduler.py:5912]: Type annotations on the `AsyncTaskGroup` are too narrow.

File: distributed/core.py [hunk 3]
```diff
@@ -673,13 +719,13 @@ def stop(self) -> None:
         self.monitor.close()
         if not (stop_listeners := self._stop_listeners()).done():
             self._ongoing_background_tasks.call_soon(
-                asyncio.wait_for(stop_listeners, timeout=None)
+                asyncio.wait_for(stop_listeners, timeout=None)  # type: ignore[arg-type]
```
Comment (hendrikmakait) [inline distributed/core.py:722]: Type annotations on the `AsyncTaskGroup` are too narrow. This should be addressed in a follow-up PR.

**Review by fjetter — APPROVED** [review]

**Comment by crusaderky** [comment]
I was too late - my review comments are now in https://github.com/dask/distributed/pull/8247

**Pull request merged**